op-e2e/actions/interop/dsl/dsl.go

```diff
@@ -374,3 +375,84 @@ func (d *InteropDSL) AdvanceL2ToLastBlockOfOrigin(chain *Chain, l1OriginHeight u
 		d.AddL2Block(chain)
 	}
 }
+
+func (d *InteropDSL) ActSyncSupernode(t helpers.Testing, opts ...actSyncSafeOption) {
+	cfg := &actSyncSafeConfig{
+		ChainOpts: d.defaultChainOpts(),
+	}
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	preSeqSyncStatuses := make([]*eth.SyncStatus, 0, len(cfg.Chains))
+	preSupSyncStatuses := make([]*eth.SupervisorChainSyncStatus, 0, len(cfg.Chains))
+	for _, chain := range cfg.Chains {
+		preSeqSyncStatus, preSupSyncStatus := d.Actors.SyncStatuses(t, chain)
+		preSeqSyncStatuses = append(preSeqSyncStatuses, preSeqSyncStatus)
+		preSupSyncStatuses = append(preSupSyncStatuses, preSupSyncStatus)
+	}
+
+	// Perform actions
+	if cfg.shouldSendL1LatestSignal {
+		d.Actors.Supervisor.SignalLatestL1(t)
+	}
+	if cfg.shouldSendL1FinalizedSignal {
+		d.Actors.Supervisor.SignalFinalizedL1(t)
+	}
+	for _, chain := range cfg.Chains {
+		// Move supervisor safe head forward by one block
+		chain.Sequencer.SyncSupervisor(t) // supervisor to react to exhaust-L1
+	}
+	d.Actors.Supervisor.ProcessFull(t)
+	for _, chain := range cfg.Chains {
+		chain.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
+	}
+
+	for i, chain := range cfg.Chains {
+		preSeqSyncStatus := preSeqSyncStatuses[i]
+		preSupSyncStatus := preSupSyncStatuses[i]
+
+		postSeqSyncStatus, postSupSyncStatus := d.Actors.SyncStatuses(t, chain)
+
+		// Post assertions
+		require.Equal(t, preSeqSyncStatus.UnsafeL2.ID(), postSeqSyncStatus.UnsafeL2.ID(), "sequencer unsafe head should not move for chain id %s", chain.ChainID)
+
+		// TODO: Add more postSupSyncStatus assertions
+		if cfg.requireLocalSafeAdvances {
+			require.Equal(t, preSeqSyncStatus.LocalSafeL2.Number+1, postSeqSyncStatus.LocalSafeL2.Number, "sequencer safe head should move forward by one block for chain id %s", chain.ChainID)
+			// No supervisor sync status localSafe field
```

Comment:
`SupervisorChainSyncStatus` contains `LocalSafe`. May I ask the intention about this comment please?
```
type SupervisorChainSyncStatus struct {
	// LocalUnsafe is the latest L2 block that has been processed by the supervisor.
	LocalUnsafe BlockRef `json:"localUnsafe"`
	LocalSafe   BlockID  `json:"localSafe"`
	CrossUnsafe BlockID  `json:"crossUnsafe"`
	// Some fault-proof releases may already depend on `safe`, so we keep JSON field name as `safe`.
	CrossSafe BlockID `json:"safe"`
	Finalized BlockID `json:"finalized"`
}
```

Reply:
Yes, the code you're referencing was added a week ago, after the introduction of this PR. This is what it looked like at the time of writing, you can see there is no supervisor sync status field:

```
type SupervisorChainSyncStatus struct {
	// LocalUnsafe is the latest L2 block that has been processed by the supervisor.
	LocalUnsafe BlockRef `json:"localUnsafe"`
	Safe        BlockID  `json:"safe"`
	Finalized   BlockID  `json:"finalized"`
}
```

https://github.com/ethereum-optimism/optimism/pull/15624/files#diff-5a46d6f6c609d5cca2cb28db80b9858873fca04a9c70f9234712f63e807f0309

I've eliminated the logic that this comment accompanies in the latest version of this PR.